drivers/gpio/gpio_mcp23xxx.h

```diff
@@ -33,18 +33,35 @@ extern "C" {
 #define REG_GPIO 0x09
 #define REG_IKAT 0x0A
 
+#define MCP23SXX_ADDR 0x40
+#define MCP23SXX_READBIT 0x01
+
+typedef int (*mcp23xxx_read_port_regs)(const struct device *dev, uint8_t reg, uint16_t *buf);
+typedef int (*mcp23xxx_write_port_regs)(const struct device *dev, uint8_t reg, uint16_t value);
+typedef bool (*mcp23xxx_bus_is_ready)(const struct device *dev);
 /** Configuration data */
-struct mcp230xx_config {
+struct mcp23xxx_config {
 	/* gpio_driver_config needs to be first */
 	struct gpio_driver_config config;
 
 	/** I2C device */
-	const struct i2c_dt_spec i2c;
+	union {
+#ifdef CONFIG_GPIO_MCP230XX
+		struct i2c_dt_spec i2c;
+#endif /* CONFIG_GPIO_MCP230XX */
+#ifdef CONFIG_GPIO_MCP23SXX
+		struct spi_dt_spec spi;
+#endif /* CONFIG_GPIO_MCP23SXX */
+	} bus;
```

Comment:
this construction implies that both can't be enabled at the same time, however, Kconfig does not reflect that. There are a few sensor drivers that support SPI and I2C, look for `DT_ANY_INST_ON_BUS_STATUS_OKAY` to find them.